packages/cli/src/commands/init.ts

```diff
@@ -268,6 +269,15 @@ const processInitForm = async (
             // noop
           }
         }
+        // If startBlock is not set, try to load it.
+        if (!startBlock) {
+          try {
+            // Load startBlock for this contract
+            startBlock = Number(await loadStartBlockForContract(network!, value)).toString();
+          } catch (error) {
+            // noop
+          }
+        }
```

Comment:
we don't follow a `retryAsync` pattern anywhere but ideally we put this logic in a retry logic and that would also mean we reflect in the prompt UI. A future improvement in CLI.

<details>
<summary>retryAsync</summary>

```ts
async function retryAsync(fn, {
  tries,
  interval,
  onError
}) {
  let currentTry = 0;
  let latestError;
  while (currentTry < tries) {
    try {
      return await fn({
        attempt: currentTry + 1
      });
    } catch (error) {
      onError == null ? void 0 : onError({ error, currentTry });
      latestError = error;
      currentTry += 1;
      if (currentTry !== tries) {
        await wait(interval);
      }
    }
  }
  throw latestError || new Error(`Invalid retries: ${tries}`);
}
```
<details>

Reply:
https://github.com/graphprotocol/graph-tooling/issues/1060

Reply:
Agree 100%, would love to work on it, i will write in the issue, would need your guidance on how to go about it.